### PR TITLE
* Don't throw if Assembly.GetEntryAssembly is null

### PR DESCRIPTION
**Pull request by pranavkm**
* Add a CompilationOptions.Default for convenient use when DependencyContext.Default is null.

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @pakrym

**Comment by pakrym** [comment]
@dotnet-bot run tests please

File: src/Microsoft.Extensions.DependencyModel/CompilationOptions.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Generic;
+using System.Linq;
```
Comment (pakrym) [inline src/Microsoft.Extensions.DependencyModel/CompilationOptions.cs]: ?

**Comment by pakrym** [comment]
:shipit:

**Pull request merged**